### PR TITLE
release note for 310

### PR DESCRIPTION
**Pull request by abby-cyber**

### CODE REVIEW

**Review by whitewum — APPROVED** [review]

File: docs-2.0/20.appendix/releasenote.md [hunk 1]
```diff
@@ -1,8 +1,58 @@
 # Nebula Graph {{ nebula.release }} release notes
 
-TODO
+## Optimization 
 
-## Legacy versions
+- Patterns can now be used in queries. For example: `MATCH (v:player) WHERE (v)-[:like]->() RETURN v`. [#3997](https://github.com/vesoft-inc/nebula/pull/3997)
```
Comment (Sophie-Xie) [inline docs-2.0/20.appendix/releasenote.md:5]: in WHERE statement

File: docs-2.0/20.appendix/releasenote.md [hunk 2]
```diff
@@ -1,8 +1,58 @@
...
+- `maxHop` is optional in MATCH variable-length paths.[#3881](https://github.com/vesoft-inc/nebula/pull/3881) 
+- Graph spaces are physically deleted after using `DROP SPACE`. [#3913](https://github.com/vesoft-inc/nebula/pull/3913) 
+- Optimized number parsing in date time, date, time. [#3797](https://github.com/vesoft-inc/nebula/pull/3797) 
+- Added the `toSet` function which converts `LIST` or `SET` to `SET`. [#3594](https://github.com/vesoft-inc/nebula/pull/3594) 
+- nGQL statements can be used to display the HTTP port of Nebula Graph services and the HTTP2 port has been disabled. [#3808](https://github.com/vesoft-inc/nebula/pull/3808) 
+- The number of sessions for connections to each graphd with the same client IP and the same user is limited. [#3729](https://github.com/vesoft-inc/nebula/pull/3729) 
+- Optimized the waiting mechanism to ensure a timely connection to the metad after the storaged starts. [#3971](https://github.com/vesoft-inc/nebula/pull/3971) 
+- It is no longer to rebuild the vertex when it has multiple paths and an error of the disk corresponding to a particular path occurs. [#4131](https://github.com/vesoft-inc/nebula/pull/4131)
```
Comment (Sophie-Xie) [inline docs-2.0/20.appendix/releasenote.md:21]: Not vertex

File: docs-2.0/20.appendix/releasenote.md [hunk 3]
```diff
@@ -1,8 +1,58 @@
...
+- The `DOWNLOAD` and `INGEST` SST files are now managed with the job manager. [#3994](https://github.com/vesoft-inc/nebula/pull/3994)
+- Support for error code display when a job fails. [#4067](https://github.com/vesoft-inc/nebula/pull/4067) 
+- The OS page cache can be disabled and the block cache and Nebula Graph storage cache can only be used in a shared environment, to avoid memory usage interference between applications. [#3890](https://github.com/vesoft-inc/nebula/pull/3890) 
+- Updated the default value of the KV separation threshold from 0 to 100. [#3879](https://github.com/vesoft-inc/nebula/pull/3879) 
+- Support for using gflag to set the upper limit of expression depth for a better fit of different machine environments. [#3722](https://github.com/vesoft-inc/nebula/pull/3722) 
+- Added a permission check for `KILL QUERY`. When the authorization is enabled, the GOD user can kill any query and the users with other roles can only kill queries that they own. [#3896](https://github.com/vesoft-inc/nebula/pull/3896)
```
Comment (Sophie-Xie) [inline docs-2.0/20.appendix/releasenote.md:28]: queries

File: docs-2.0/20.appendix/releasenote.md [hunk 4]
```diff
@@ -1,8 +1,58 @@
...
+- Fixed the crash when the `async` interface in the Lambda function finished running and the task in `threadManager` was not executed. [#4000](https://github.com/vesoft-inc/nebula/pull/4000) 
+- Fixed the `GROUP BY` output bug. [#4128](https://github.com/vesoft-inc/nebula/pull/4128) 
+- Fixed the bug that the version wasn't displayed with `SHOW HOSTS` sometimes. [#4116](https://github.com/vesoft-inc/nebula/pull/4116) 
+- Fixed the bug on parameters for `id(n) == $var`, `id(n) IN [$var]`, `id(n) == $var.foo.bar`, and `id(n) IN $var.foo.bar`. [#4024](https://github.com/vesoft-inc/nebula/pull/4024) 
+- Fixed the bug that an incorrect path direction occurred in `MATCH...WHERE`. [#4091](https://github.com/vesoft-inc/nebula/pull/4091) 
+- Fixed the bug that the result of referencing multiple `MATCH` variables in a `WHERE` clause was incorrect. [#4143](https://github.com/vesoft-inc/nebula/pull/4143) 
+- Fixed the optimizer bug. [#4146](https://github.com/vesoft-inc/nebula/pull/4146) 
+- Fixed the bug that the storage service failed to handle Raft snapshots. [#4019](https://github.com/vesoft-inc/nebula/pull/4019)
```
Comment (Sophie-Xie) [inline docs-2.0/20.appendix/releasenote.md:46]: node

Comment (abby-cyber) [reply docs-2.0/20.appendix/releasenote.md:46]: maybe we can use service here instead of node.

**Review by cooper-lzy — APPROVED** [review]

**Review by randomJoe211 — APPROVED** [review]

**Pull request merged**